docs/INSTALL.md

```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
+* Update mode for devices: `dpdk`, `af_xdp` or `af_packet` in [`scripts/docker_setup.sh`](../scripts/docker_setup.sh), along with device details
+  - If planning to use DPDK, follow [these](dpdk-configuration.md) directions before moving to the next steps
+
+>`./scripts/docker_setup.sh` is a quick start guide to set up UPF for evaluation.
 
->`scripts/docker_setup.sh` is a quick start guide to set up UPF for evaluation.
+## Installation Simulation mode
+
+To install the UPF in simulation mode, the following changes are required:
+
+1. Enable sim mode in configuration file
+
+    ```patch
+    diff --git a/conf/upf.json b/conf/upf.json
+    index c6531cc..928a503 100644
+    --- a/conf/upf.json
+    +++ b/conf/upf.json
+    @@ -2,8 +2,8 @@
+         "": "Vdev or sim support. Enable `\"mode\": \"af_xdp\"` to enable AF_XDP mode, or `\"mode\": \"af_packet\"` to enable AF_PACKET mode, `\"mode\": \"sim\"` to generate synthetic traffic from BESS's Source module or \"mode\": \"\" when running with UP4",
+         "": "mode: af_xdp",
+         "": "mode: af_packet",
+    -    "": "mode: sim",
+    -    "mode": "dpdk",
+    +    "mode": "sim",
+    +    "": "mode: dpdk",
 
-## Run
+         "table_sizes": {
+             "": "Example sizes based on sim mode and 50K sessions. Customize as per your control plane",
+    ```
+
+2. Enable sim mode in script file
+    ```patch
+    diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+    index 086ad2f..79d81bd 100755
+    --- a/scripts/docker_setup.sh
+    +++ b/scripts/docker_setup.sh
+    @@ -16,7 +16,7 @@ bessd_port=10514
+    mode="dpdk"
+    #mode="af_xdp"
+    #mode="af_packet"
+    -#mode="sim"
+    +mode="sim"
+
+    # Gateway interface(s)
+    #
+    ```
+
+3. Start installation from UPF's root directory
+
+    ```bash
+    ./scripts/docker_setup.sh
+    ```
+
+4. Insert rules into PDR and FAR tables
+
+    Use gRPC sim mode to directly install PFCP forwarding rules via gRPC API (works only for BESS)
+
+    ```bash
+    docker exec -ti bess-pfcpiface pfcpiface -config /conf/upf.json -simulate create
+    ```
+
+    OR
+
+    Use the [pfcpsim](https://github.com/omec-project/pfcpsim) tool to generate PFCP messages towards the PFCP Agent.
+
+5. (optional) Observe the pipeline in GUI
+
+   To display GUI of the pipeline visit [http://[hostip]:8000](http://[hostip]:8000)
+
+
+## Installation DPDK mode
+
+To install the UPF in DPDK mode, the following changes are required:
+
+1. Make appropriate changes in configuration file
+
+    ```patch
+    $ git diff conf/upf.json
+    diff --git a/conf/upf.json b/conf/upf.json
+    index c6531cc..335e33b 100644
+    --- a/conf/upf.json
+    +++ b/conf/upf.json
+    @@ -51,7 +51,7 @@
+         "gtppsc": false,
+
+         "": "Enable Intel Dynamic Device Personalization (DDP)",
+    -    "ddp": false,
+    +    "ddp": true,
+
+         "": "Telemetrics-See this link for details: https://github.com/NetSys/bess/blob/master/bessctl/module_tests/timestamp.py",
```

Comment:
Does this change now given we have bess repository forked ?

Reply:
Yes, I am going to add another commit changing all the references to the new repo/fork.

Reply:
I just added another commit to this PR to address your comment. However, there are still two places in the code with links to the original BESS. One is in `Dockerfile` that is address in #637 and in `pfcpiface/bess_pb/module_msg.pb.go` that needs to be changed/updated in the BESS fork and regenerate the GO code with the updated links